lib/scheduler.py

```diff
@@ -339,15 +339,18 @@ def _task(self, name, obj, by, source, dest, value):
             sh = self._sh  # noqa
             try:
                 exec(obj.bytecode)
+                for method in logic.get_method_triggers():
```

Comment:
I'm not sure if this is a bug or a feature :smiley: :
Due to the try/except Block starting in line 340, the hooked methods will only be called if the logic could be executed without error, and nobody called "exit()" inside the logic. In case of errors or "exit()" in the logic, the hooked methods will not be called.